model/profile/_meta/fields.yml

```diff
@@ -0,0 +1,118 @@
+- key: apm-profile
+  title: APM Profile
+  description: Profiling-specific data for APM.
+  fields:
+    - name: profile
+      type: group
+      dynamic: false
```

Comment:
I don't see any restrictions on the Intake API for which profile values can be processed (cpu, in_use, etc), as long as they adhere to the `google/pprof/profile` structure. There is also no restrictions on the model transformation. This is dynamic and open for introducing more profile values in the future, but the ES template only indexes a defined set of values, meaning we would still need to adapt the server when additional values should be indexed. 

This might be perfectly fine and on purpose, but we might also want to take an approach similar to what we do with labels, where we dynamically index everything that adheres to a defined format.

Reply:
The reason I went with this is so that we could have well-defined metric names for the UI to look for. It's possible we'll want to accept other profile formats, which are unlikely to use the same names, so we would map to these names.

Making it dynamic sounds reasonable. I'm a bit concerned about making the `profile` group dynamic, so I suppose this would require adding another level to the metric names.

As this is currently only intended for the server profiling itself, and at the moment these are the only fields that will come in via the profiles we gather, I'm inclined to defer deciding on this. The whole thing is experimental so we can make changes along the way.